### ISSUE TITLE
NCAUpdate import error in PyPi version of package

### ISSUE THREAD

**Issue by JELAshford** [maxencefaldor/cax#3]
Hi! Love the package and paper, and wanted to reproduce some of the results using one of the Google Colabs. 

I ran into an issue with the [self-classifying MNIST digits](https://colab.research.google.com/github/maxencefaldor/cax/blob/main/examples/self_classifying_mnist.ipynb) example, where "Run All" fails at the import cells when importing `NCAUpdate`.

I've installed the package locally from `pip` and the `mlp_update` file in `cax.core` is different on the `pip` installable version to the current GitHub version. It looks like at some point the `.nnx` import has been duplicated, resulting in `from flax.nnx.nnx.nn import initializers` which fails. 

Steps for a minimal repro with the `uv` package manager here: 
```bash
> uv init cax-test --python=3.12
Initialized project `cax-test` at `/Users/jamesashford/Downloads/cax-test`
> cd cax-test
> uv add cax -q
> source .venv/bin/activate
> python
Python 3.12.7 (main, Oct  1 2024, 02:05:46) [Clang 15.0.0 (clang-1500.3.9.4)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from cax.core.update.nca_update import NCAUpdate

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/jamesashford/Downloads/cax-test/.venv/lib/python3.12/site-packages/cax/core/update/nca_update.py", line 10, in <module>
    from cax.core.update.residual_update import ResidualUpdate
  File "/Users/jamesashford/Downloads/cax-test/.venv/lib/python3.12/site-packages/cax/core/update/residual_update.py", line 7, in <module>
    from cax.core.update.mlp_update import MLPUpdate
  File "/Users/jamesashford/Downloads/cax-test/.venv/lib/python3.12/site-packages/cax/core/update/mlp_update.py", line 7, in <module>
    from flax.nnx.nnx.nn import initializers
ModuleNotFoundError: No module named 'flax.nnx.nnx'
>>> quit()
> 
```

The version of the `mlp_update` file on GitHub doesn't have this issue, and installing directly from GitHub (with `pip install --upgrade git+https://github.com/maxencefaldor/cax.git`) works as expected. So, it's possible that the `cax` version on `PyPi` needs to be updated to reflect the current GitHub version? Happy to provide more info if needed!

**Comment by maxencefaldor** [comment]
Hi,

I am glad to hear that you love CAX! I really appreciate you taking the time to open this issue.

I've just released version 0.1.6 on PyPI which fixes the `flax.nnx` import issue you encountered.

**Comment by JELAshford** [comment]
Can confirm that this is working on my end now - thank you very much! Closing as fixed :)
